pkg/utils/path.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package utils
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+type FPath struct {
+	local  bool
+	schema string
+	path   string
+}
+
+// Creates new Struct from handed over URL
+func (p *FPath) New(url string) {
+
+	// Check for Schema
+	sepstring := strings.Split(url, "://")
+
+	switch len(sepstring) {
+	case 2: // Has Schema
+		p.schema = sepstring[0]
+		p.path = sepstring[1]
+	case 1: // No Schema
+		p.local = true
+		p.path = sepstring[0]
+	default: // Everything else is misformatted
+		log.Fatalf("misformatted URL: %v", url)
+	}
+
+	// Check for Windows Special Handling Prefix
+	cprefix, _ := regexp.Compile(`^\\\\\?\\(UNC\\)?`)
```

Comment:
Use `string.HasPrefix` not regexps.

Reply:
I am using the RegExp to catch both variants that are allowed (with and without UNC\) and select that part directly.
strings package doesnt have regExp does it?

Reply:
You can take a look what https://golang.org/src/path/filepath/path_windows.go does